### ISSUE TITLE
Hub-detect doesn't delete old code locations from a scan

### ISSUE THREAD

**Issue by imbeyondboredom** [blackducksoftware/hub-detect#229]
I've found that when you set projectCodeLocationDeleteOldNames to true is does not carry down to the Scanner. Here are the logs from my build:
```
[22:10:15][Step 1/1] policyCheck = false
[22:10:15][Step 1/1] projectCodeLocationDeleteOldNames = true
[22:10:15][Step 1/1] projectLevelMatchAdjustments = true
...
[22:10:20][Step 1/1] 2018-04-10 22:10:20 INFO  [pool-2-thread-1] --- --> Un-map previous Code Locations : false
[22:10:20][Step 1/1] 2018-04-10 22:10:20 INFO  [pool-2-thread-1] --- --> Delete previous Code Locations : false
```

I believe you just need to carry the setting for deleting code locations down to this point and then call setDeletePreviousCodeLocations. It's possible this should also call setUnmapPreviousCodeLocations?

https://github.com/blackducksoftware/hub-detect/blob/ef7ed2c4708cdcc4a6faf6f41a88c2934a3fa33f/src/main/groovy/com/blackducksoftware/integration/hub/detect/hub/HubSignatureScanner.java#L269

**Comment by jamesrichard91** [comment]
I will agree the logs are confusing and we should probably address that. That being said, 
[22:10:15][Step 1/1] projectCodeLocationDeleteOldNames = true
and 
[22:10:20][Step 1/1] 2018-04-10 22:10:20 INFO  [pool-2-thread-1] --- --> Delete previous Code Locations : false

Are actually 2 completely different things.  The first is the property for Detect to delete ONLY the code locations created by previous runs of Detect that are mapped to the current Project Version in the Hub that you are adding code locations to.  The second is a general setting that we created for a different Integration that will delete ALL code locations that are mapped to the current Project Version in the Hub that you are adding code locations to.

**Comment by imbeyondboredom** [comment]
Ah - so it appears that I hit an edge case. So I was scanning this old repo of mine to test and I scanned these two branches back to back under the same project/
version:
https://github.com/imbeyondboredom/dependency-check-gradle/tree/master
https://github.com/imbeyondboredom/dependency-check-gradle/tree/test/blackduck

In the test/blackduck branch the code was moved down a directory and when that happened hub-detect didn't replace the old scans. 

While it's odd to scan two different branches with the same version - If the test/blackduck branch was merged into the master branch then that old scan would stick around.

<img width="1388" alt="screen shot 2018-04-12 at 10 50 31 pm" src="https://user-images.githubusercontent.com/86616/38714451-54a5a462-3ea4-11e8-9aa7-5823b3f0e09e.png">
